components/Image.js

```diff
@@ -23,16 +23,19 @@ class Image extends Component {
 
     // defaultSource is not supported on Android, so we manually
     // reassign the defaultSource prop to source if source is not defined
-    if ((Platform.OS === 'android') && (!source || !source.uri)) {
-      resolvedProps = {
-        ...props,
-        // Image views are not rendered on Android if there is no image to display,
-        // so we fallback to a transparent image to be compatible with iOS
-        source: defaultSource || require('../assets/images/transparent.png'),
-        // Fixes a bug with local image resizing on Android:
-        // https://github.com/facebook/react-native/issues/4598#issuecomment-162328501
-        style: { ...props.style, width: null },
-      };
+    if (Platform.OS === 'android') {
+      // a valid source is either an object with an uri key or a number (from a `require` call)
+      if (typeof source === 'object' ? !source.uri : typeof source !== 'number') {
```

Comment:
we use lodash as dependency, so I assume you could write this to look more elegant:
```javascript
if (_.isNumber(source) || (_.isObject(source) && source.uri))
```